var/spack/repos/builtin/packages/font-util/package.py

```diff
@@ -102,6 +102,8 @@ def font_install(self):
 
         for font in fonts:
             fontroot = find(font, '*', recursive=False)
+            if not fontroot:
+                continue
```

Comment:
Ideally if a variant requests something, we should always provide it or fail. Is it possible that some fonts are missing on certain systems? In that case, would it be possible to edit the construction of `fonts` (up around line 70) to exclude certain fonts for certain systems?

Or perhaps we can split fonts into two variants: one set that we expect to be everywhere, and a set of "extended" fonts that are of interest but might not be available everywhere.

Reply:
@bernhardkaindl do you think you'll be able to check into this in the next week or so?

Reply:
I agree with the principle that the installation of a font should not fail silently.

But, I cannot implement your specific suggestion anymore however:

The system where the install failed now works. It wasn't upgraded but some packages were installed,
and it could be that ca-certificates were updated since that as well.

Thus, I think that editing the list of fonts depending on the platform or distribution does not get to the root of the cause. Measures like this which are in the package currently seem to have a different root, but I think they illustrate the issue:
```py
        conflicts('fonts=font-bh-ttf', when='platform=cray')
        conflicts('fonts=font-bh-ttf', when='arch=linux-rhel7-broadwell')
```
What's missing here is an explanation what e.g. cray or especially a system using an Intel broadwell CPU has to do with of a font can be installed. There can be none. Therefore I think also that the logic of your proposal does not get to the root of why the installation failed on that system.

What should be done however is: When installing a font fails, it should be shown which font failed.

As the `find` function failed to find the font's source directory in my case, there must have been some error downloading or extracting the font's source, which could have been caused by the ca-certificates possibly needing an update at that time, and then not catching such error for the `resource` downloaded.

At least that is the only explanation I could come up with now.